core/sr-api-macros/src/impl_runtime_apis.rs

```diff
@@ -377,6 +379,47 @@ fn generate_runtime_api_base_structures(impls: &[ItemImpl]) -> Result<TokenStrea
 				res
 			}
 
+			fn call_api_at_with_context<
```

Comment:
Just take the context as argument to `call_api_at`. And use `Other` as default for each call that is not a `_with_context` call.